### PR TITLE
[BTS-491][3.8] ArangoRestore connect Retry

### PR DESCRIPTION
**Pull request by maierlars**
### Scope & Purpose
Forward port of https://github.com/arangodb/arangodb/pull/14437.

### CODE REVIEW

**Comment by maierlars** [comment]
https://jenkins.arangodb.biz/job/arangodb-matrix-pr/16066/

File: arangosh/Restore/RestoreFeature.cpp [hunk 1]
```diff
@@ -1539,6 +1539,13 @@ void RestoreFeature::collectOptions(std::shared_ptr<options::ProgramOptions> opt
                   new UInt32Parameter(&_options.threadCount))
       .setIntroducedIn(30400);
 
+  options
+      ->addOption("--initial-connect-retries",
+          "number of connect retries for initial connection",
+          new UInt32Parameter(&_options.initialConnectRetries))
+      .setIntroducedIn(30800)
```
Comment (jsteemann) [inline arangosh/Restore/RestoreFeature.cpp:1546]:
```suggestion
      .setIntroducedIn(30801)
```

Comment (jsteemann) [reply arangosh/Restore/RestoreFeature.cpp:1546]: @Simran-B : what is our convention here? Should we actually list previous versions here too (i.e. mention `30713` **and** `30801`, or should we just mention `30801` in 3.8?)

File: CHANGELOG [hunk 2]
```diff
@@ -1,5 +1,9 @@
 v3.8.0 (XXXX-XX-XX)
--------------------
+--------------------
```
Comment (jsteemann) [inline CHANGELOG:2]:
```suggestion
-------------------
```

File: CHANGELOG [hunk 3]
```diff
@@ -1,5 +1,9 @@
 v3.8.0 (XXXX-XX-XX)
```
Comment (jsteemann) [inline CHANGELOG:1]: The feature will get only into 3.8.1, so the CHANGELOG needs adjustment before merging.

**Pull request merged**